File: sol/solution/0001-0100/0001. Two Sum/Solution.ts
function twoSum(nums: number[], target: number): number[] {
    const hashMap = {};
    for(let i = 0 ; i < nums.length; i++){
        if(target-nums[i] in hashMap){
            return [hashMap[target-nums[i]],i];
        }
        hashMap[nums[i]]=i;
    }
    return [-1,-1];
};

File: sol/solution/0001-0100/0001. Two Sum/Solution.erl
-spec two_sum(Nums :: [integer()], Target :: integer()) -> [integer()].
two_sum(Nums, Target) ->
    find_target_sum(Nums, Target, 0, #{}).
find_target_sum([Num | NextNums], Target, Index, Map) ->
    NumInMap = maps:is_key(Num, Map),
    if 
        NumInMap ->
            [maps:get(Num, Map), Index];
        true ->
            find_target_sum(NextNums, Target, Index+1, maps:put(Target-Num, Index, Map))
    end.